hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1439,10 +1439,13 @@ public S3Object getObject(GetObjectRequest request) {
    * using FS state as well as the status.
    * @param fileStatus file status.
    * @param seekPolicy input policy for this operation
+   * @param changePolicy change policy for this operation.
    * @param readAheadRange readahead value.
+   * @param auditSpan audit span.
    * @return a context for read and select operations.
    */
-  private S3AReadOpContext createReadContext(
+  @VisibleForTesting
+  protected S3AReadOpContext createReadContext(
```

Comment:
I'm not really convinced that this is needed. Check the main comment for details.

Reply:
Since the read context attach the retry policy and the retry logic that being used by the input stream, this is seems to be the entry where we can get for the test without exposing many internal implementations.

Reply:
Afraid we currently do. However, if we move to a builder API for that ReadOpContext then the test could construct something very minimal (would only need the Invoker ref). I'd support that change here as it would help future tests.